src/simulation/LoadGenerator.cpp

```diff
@@ -244,13 +291,46 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
             MIN_UNIQUE_ACCOUNT_MULTIPLIER);
     }
 
-    if (cfg.mode == LoadGenMode::SOROBAN &&
+    if ((cfg.mode == LoadGenMode::SOROBAN_INVOKE ||
+         cfg.mode == LoadGenMode::SOROBAN_UPLOAD ||
+         cfg.mode == LoadGenMode::SOROBAN_INVOKE_SETUP) &&
         protocolVersionIsBefore(mApp.getLedgerManager()
                                     .getLastClosedLedgerHeader()
                                     .header.ledgerVersion,
                                 SOROBAN_PROTOCOL_VERSION))
     {
-        errorMsg = "Soroban mode requires protocol version 20 or higher";
+        errorMsg = "Soroban modes require protocol version 20 or higher";
+    }
+
+    if (cfg.mode == LoadGenMode::SOROBAN_INVOKE)
+    {
+        auto& sorobanCfg = mApp.getLedgerManager().getSorobanNetworkConfig();
+        if (mContractInstances.size() < cfg.nAccounts)
```

Comment:
I'm not sure this check makes sense: mContractInstances is populated during loadgen, whereas nAccounts is const config, so mContractInstances should depend on nAccounts to begin with

Reply:
The issue is that `mContractInstances` is populated during `SOROBAN_INVOKE_SETUP`, but used during `SOROBAN_INVOKE`. It's possible that these get run with different configurations. This is an error check for situations where `nAccounts == 10` during setup loadgen but `nAccounts == 20` during the invoke loadgen.